slimdiffy/autodiff.py

```diff
@@ -1004,8 +1015,11 @@ def visit_var(self, eq: Var, derivative: Tracer) -> None:
         pass
 
     def visit_add(self, eq: Op, derivative: Tracer) -> None:
-        self.derivatives[eq.inputs[0]] += derivative
-        self.derivatives[eq.inputs[1]] += derivative
+        for input_idx in eq.inputs:
+            if isinstance(derivative, pt.Node):
+                derivative = derivative.leaf_value
+            # For addition, derivative wrt each input is just the derivative
+            self.derivatives[input_idx] = _ensure_tracer(derivative, self.supervisor)
```

Comment:
Hmm I don't think this is needed? The derivative should never be a pt.Node. Tracer results are *always* tensors, never tuples.

Also lets not add this loop. I think you can revert changes to this function.